### PR TITLE
ISPN-11615 Replace internal metadata class

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-2634

Depends on:
https://github.com/infinispan/infinispan/pull/8113
https://github.com/infinispan/infinispan/pull/8180

### CODE REVIEW

**Comment by ryanemerson** [comment]
@pruivo  Thanks for getting this up so quick. One quick observation before I do a proper review. Personally I think the name `InternalMetadata` is a better fit, but I know we already have the `org.infinispan.metadata.InternalMetadata` interface. If we wanted to stick with this name, we can remove the old `InternalMetadata` as it was deprecated in 10.0.

**Comment by ryanemerson** [comment]
I've created https://issues.redhat.com/browse/ISPN-11617 if you're interested, but does seem like `InternalMetadataImpl` is still used by the `CallIntercepter` for solving https://issues.redhat.com/browse/ISPN-4075. \\cc @danberindei

**Comment by pruivo** [comment]
@ryanemerson InternalMetadata is still used almost "everywhere". That's why I created "Private" metadata :)

File: commons/all/src/main/java/org/infinispan/commons/marshall/ProtoStreamTypeIds.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
    int META_PARAMS_INTERNAL_METADATA = CORE_LOWER_BOUND + 3;
    int REMOTE_METADATA = CORE_LOWER_BOUND + 4;
    int UUID = CORE_LOWER_BOUND + 5;
+   int PRIVATE_METADATA = UUID + 1;
```
Comment (ryanemerson) [inline commons/all/src/main/java/org/infinispan/commons/marshall/ProtoStreamTypeIds.java:56]: We probably want this in the `0..127` range as it is included in so many commands.

Comment (pruivo) [reply commons/all/src/main/java/org/infinispan/commons/marshall/ProtoStreamTypeIds.java]: done! 
is there a special case for ids < 128?

Comment (ryanemerson) [reply commons/all/src/main/java/org/infinispan/commons/marshall/ProtoStreamTypeIds.java]: Ids < 128 only require a single byte to marshall. But obviously we only have a limited number of messages that can be in this range, so we should save this range for protobuf messages which are sent frequently

**Comment by ryanemerson** [comment]
Thanks @pruivo

**Pull request merged**